### DISCUSSION TITLE
write JSON with syntax highlighting

### DISCUSSION THREAD

**Discussion by thoemmi** [spectreconsole/spectre.console#820] [Ideas]
Some time ago I wrote an extension for Spectre.Console to write JSON with syntax highlighting. Now I'm writing my third CLI where I'm using that code, and I'm tired of copying it again and again :wink: Maybe the code is useful enough to be added to Spectre.Console. I'd be happy to contribute.

E.g. here's some sample code:

```csharp
var obj = new
{
    Text = "Hello",
    Flag = true,
    Numbers = new[] { 1, 2, 3 },
    Objects = new[] { new { Name = "Foo" }, new { Name = "Bar" } },
    Nested = new
    {
        NestedText = "World",
        Number = 42
    },
    Dictionary = new Dictionary<string, string> { { "Key1", "Value1" }, { "Key2", "Value2" } },
    NullObject = (object?)null,
    EmptyObject = new{}
};

var jsonElement = JsonSerializer.SerializeToElement(obj);

AnsiConsole.Console.WriteJson(jsonElement);
```

produces

![image](https://user-images.githubusercontent.com/73690/165942524-73af4acb-728f-48d4-bf4d-e06e41e7c0d1.png)

**Comment by patriksvensson** [comment]
I think this would be useful, but I'm a bit hesitant adding a dependency on a JSON library, to be honest.
Is there any way we could render JSON without depending on a JSON library?

**Reply by pmengal** [reply]
Can we done without it. Check this project. https://github.com/goblinfactory/pretty-json

**Reply by thoemmi** [reply]
@pmengal pretty-json depends on System.Text.Json :wink:

**Reply by pmengal** [reply]
Which is a part of the framework. Same status as System.Memory for example.

**Reply by thoemmi** [reply]
System.Text.Json is included in the .NET Core runtime since 3.1. If @patriksvensson hesitates to add the dependency for .NET Standard 2.0, I could just publish a separate package (somewhat overkill for just one small file though) or add it to the documentation.

**Comment by olabacker** [comment]
Mind sharing the extension? :)

**Reply by thoemmi** [reply]
No problem. See this gist: https://gist.github.com/thoemmi/150ae4622ea651f6856b92703b8e0f7a

**Reply by olabacker** [reply]
Thanks alot! Very kind 😄
